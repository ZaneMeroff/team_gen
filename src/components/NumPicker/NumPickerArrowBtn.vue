<template>

  <button @click="emitArrowBtnClick" class="np-arrow-btn">
    {{ arrowDirection === "up" ? "&#8593" : "&#8595" }}
  </button>

</template>

<script>
  export default {
    name: "NumPickerArrowBtn",
    props: {
      arrowDirection: { type: String, default: "up" },
    },
    components: {},
    data() {
      return {

      }
    },
    methods: {

      emitArrowBtnClick() {
        this.$emit("arrowClick", this.arrowDirection )
      },
    },
    computed: {},
    watch: {},
  }
</script>

<style scoped>

  .np-arrow-btn {
    height: 75px;
    width: 75px;
    display: flex;
    justify-content: center;
    align-items: center;
    font-size: 3rem;
  }

  .np-arrow-btn:hover {
    cursor: pointer;
  }
</style>
