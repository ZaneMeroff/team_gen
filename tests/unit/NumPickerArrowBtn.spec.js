import { shallowMount } from '@vue/test-utils'

import NumPickerArrowBtn from "@/components/NumPicker/NumPickerArrowBtn"

describe("NumPickerArrowBtn", () => {

  describe("snapshots", () => {

    it("should render with default props", () => {
      const component = shallowMount(NumPickerArrowBtn)

      expect(component.element).toMatchSnapshot()
    })

    it("should set arrowDirection when prop is passed", () => {
      const propsData = { arrowDirection: "down" }
      const component = shallowMount(NumPickerArrowBtn, { propsData })

      expect(component.element).toMatchSnapshot()
    })
  })

  describe("props", () => {

    describe("arrowDirection", () => {

      it("should have a default value", () => {
        const component = shallowMount(NumPickerArrowBtn)

        expect(component.vm.arrowDirection).toEqual("up")
      })

      it("should accept a string", () => {
        const propsData = { arrowDirection: "down" }
        const component = shallowMount(NumPickerArrowBtn, { propsData })

        expect(component.vm.arrowDirection).toEqual(propsData.arrowDirection)
      })
    })
  })

  describe("methods", () => {

    describe("emitArrowBtnClick", () => {

      it("should emit arrowClick with correct payload", () => {
        const component = shallowMount(NumPickerArrowBtn)

        component.vm.emitArrowBtnClick()

        const expected = { "arrowClick": [[]] }
        expect(component.emitted()).toEqual(expected)
      })
    })
  })

  describe("user interactions", () => {

    describe("events", () => {

      it("should emit arrowClick when button is clicked", () => {
        const component = shallowMount(NumPickerArrowBtn)

        component.find(".np-arrow-btn").trigger("click")

        const expected = { "arrowClick": [[]] }
        expect(component.emitted()).toEqual(expected)
      })
    })
  })
})
