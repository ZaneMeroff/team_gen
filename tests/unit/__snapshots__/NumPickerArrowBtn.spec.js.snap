// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumPickerArrowBtn snapshots should render with default props 1`] = `
<button
  class="np-arrow-btn"
>
  
  ↑

</button>
`;

exports[`NumPickerArrowBtn snapshots should set arrowDirection when prop is passed 1`] = `
<button
  class="np-arrow-btn"
>
  
  ↓

</button>
`;
